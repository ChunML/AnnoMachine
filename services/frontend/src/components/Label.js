import React from 'react';
import PropTypes from 'prop-types';

function Label({ label }) {
  return <span className="box-label">{label}</span>;
}

Label.propTypes = {
  label: PropTypes.string.isRequired,
};

export default Label;
