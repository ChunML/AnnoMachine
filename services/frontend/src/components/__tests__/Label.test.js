import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Label from '../Label';
import '../../setupTests';

const label = 'dog';

it('Label renders properly', () => {
  const wrapper = shallow(<Label label={label} />);
  const spans = wrapper.find('span');
  expect(spans.length).toBe(1);
  expect(spans.text()).toEqual('dog');
});

it('Label renders a snapshot properly', () => {
  const tree = renderer.create(<Label label={label} />).toJSON();
  expect(tree).toMatchSnapshot();
});
