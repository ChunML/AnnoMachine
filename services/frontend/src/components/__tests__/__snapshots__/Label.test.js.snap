// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders a snapshot properly 1`] = `
<span
  className="ui header"
  style={
    Object {
      "color": "firebrick",
      "letterSpacing": "0.25rem",
    }
  }
>
  dog
</span>
`;
